### ISSUE TITLE
SC2181 argues even if command is excessively long

### ISSUE THREAD

**Issue by Starttoaster** [koalaman/shellcheck#2465]
#### For bugs
- Rule Id (if any, e.g. SC1000):  SC2181
- My shellcheck version (`shellcheck --version` or "online"):  
- [x] The rule's wiki page does not already cover this (e.g. https://shellcheck.net/wiki/SC2086)
- [x] I tried on https://www.shellcheck.net/ and verified that this is still a problem on the latest commit

#### For new checks and feature suggestions
- [x] https://www.shellcheck.net/ (i.e. the latest commit) currently gives no useful warnings about this
- [x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/bash

curl -X POST --fail \
--cert "/path/to/some-cert-file.crt" \
--key "/path/to/some-key-file.key" \
-d '{}' -k -H "Content-Type: application/json" https://localhost:port-number/route
    
if [[ "$?" -ne 0 ]]; then
  echo "$(date -u) healthcheck failed"
  exit 1
fi
```

#### Here's what shellcheck currently says:
Check exit code directly with e.g. 'if ! mycmd;', not indirectly with $?.

#### Here's what I wanted or expected to see:
Nothing. Because the command that I could directly check would need to exist on one large line that would then look ugly. At its root, this shellcheck rule is a style guideline, and it may directly conflict with other style guidelines.

**Comment by TinCanTech** [comment]
There are many ways you can write your code to comply with `shellcheck` or you can simply ignore it.

**Comment by brother** [comment]
Totally agree with @TinCanTech  here. A good case for a ignore line, shellcheck is a good friend and a guide - not a judge or headsman that you need to fear having a different opinion.

**Comment by Starttoaster** [comment]
> Totally agree with @TinCanTech  here. A good case for a ignore line, shellcheck is a good friend and a guide - not a judge or headsman that you need to fear having a different opinion.

I agree. It just feels sloppy to me adding shellcheck ignore lines for various rules I disagree with, and then need to also document the reason I added that shellcheck ignore line to cover my ass when people are code reviewing. Also many people use shellcheck in CI which isn't able to interpret nuance such as why I situationally disagree with their ruling, which means I'm forced to just use shellcheck ignores, which is unideal to me for reasons mentioned.

Anyway I created this Issue because I saw in other Issues these fairly opinionated rules got refined based off of community feedback. If Shellcheck is currently perfect and no longer needs adjusting, I rescind my comments 😜

**Comment by TinCanTech** [comment]
I look at it this way:
- If I'm **not** going to change it then I will ignore `shellcheck`
- If I'm going to change it then I let `spellcheck` complain, until I change it.
